src/relay.js

```diff
@@ -107,6 +107,127 @@ export function nodeType(connectionType) {
   return connectionType._fields.edges.type.ofType._fields.node.type;
 }
 
+function orderByAttribute(orderAttr, {source, args, context, info}) {
+  return typeof orderAttr === 'function' ? orderAttr(source, args, context, info) : orderAttr;
+}
+
+function getOrder(orderBy, options) {
+  const {model} = options;
+  const result = orderBy.map(([orderAttr, orderDirection]) => [
+    orderByAttribute(orderAttr, options),
+    orderDirection,
+  ]);
+  model.primaryKeyAttributes.forEach(primaryKeyAttribute => {
+    if (result.findIndex(([attr]) => attr === primaryKeyAttribute) < 0) {
+      result.push([primaryKeyAttribute, 'ASC']);
+    }
+  });
+  return result;
+}
+
+function reverseDirection(direction) {
+  return direction.indexOf('ASC') >= 0
+    ? direction.replace('ASC', 'DESC')
+    : direction.replace('DESC', 'ASC');
+}
+
+function reverseOrder(order) {
+  return order.map(([orderAttr, orderDirection]) => [orderAttr, reverseDirection(orderDirection)]);
+}
+
+
+/**
+ * Creates a cursor given a node returned from the Database
+ * @param  {Object}   node            sequelize model instance
+ * @param  {String[]} orderAttributes  the attributes pertaining in ordering
+ * @return {String}                   The Base64 encoded cursor string
+ */
+function toCursor(node, info) {
+  return base64(JSON.stringify(info.orderAttributes.map(attr => node.get(attr))));
+}
+
+/**
+ * Creates a cursor given a node returned from the Database
+ * (in the case that an offset had to be used instead of a window)
+ * @param  {Object}   node            sequelize model instance
+ * @param  {String[]} orderAttributes  the attributes pertaining in ordering
+ * @return {String}                   The Base64 encoded cursor string
+ */
+function toOffsetCursor(node, index) {
+  const {primaryKeyAttribute} = getModelOfInstance(node);
+  const id = typeof primaryKeyAttribute === 'string' ? node.get(primaryKeyAttribute) : null;
+  return base64(JSON.stringify([id, index]));
+}
+
+/**
+ * Decode a cursor into its component parts
+ * @param  {String} cursor Base64 encoded cursor
+ * @return {any[]}         array containing values of attributes pertaining to ordering
+ */
+function fromCursor(cursor) {
+  return JSON.parse(unbase64(cursor));
+}
+
+const dialectsThatSupportTupleComparison = {
+  mysql: true,
+  postgres: true,
+  sqlite: true,
+};
+
+function tupleComparison(model, attributes, inequality, values) {
+  const {sequelize, QueryGenerator} = model;
+  const attributesStr = attributes.map(attribute => QueryGenerator.quoteIdentifier(attribute)).join(', ');
+  const escapeOptions = {context: 'SELECT'};
+  const valuesStr = attributes.map((attribute, i) =>
+    QueryGenerator.escape(values[i], model.attributes[attribute], escapeOptions)
+  ).join(', ');
+  return sequelize.literal(`(${attributesStr}) ${inequality} (${valuesStr})`);
+}
+
+function getWindow({model, cursor, order, inclusive}) {
```

Comment:
can you provide an example of what SQL this would generate?

Reply:
Basically `(name, id) < ('foo', 5)` for instance

Reply:
Instead of `name < 'foo' OR (name = 'foo' AND id < 5)`